### PR TITLE
Deploy with data extension

### PR DESCRIPTION
**Pull request by jao**
send email, and data extension using fuel sdk

### CODE REVIEW

File: exact_target_endpoint.rb [hunk 1]
```diff
@@ -6,10 +10,17 @@ class ExactTargetEndpoint < EndpointBase::Sinatra::Base
   Honeybadger.configure do |config|
     config.api_key = ENV['HONEYBADGER_KEY']
     config.environment_name = ENV['RACK_ENV']
-  end
+  end if ENV['HONEYBADGER_KEY'].present?
```
Comment (alxpborges) [inline exact_target_endpoint.rb]: Is the Nginx passing this variable  on production environment?

File: spec/exact_target_endpoint_spec.rb [hunk 2]
```diff
@@ -1,47 +1,66 @@
...
+  context 'send_email' do
+    let(:message) {
+      Factory.payload.merge({:parameters => Factory.config}).merge(Factory.email)
+    }
+
+    it "sends e-mail" do
+      VCR.use_cassette('send_email_success') do
+        post '/send_email', message.to_json, auth
+
+        binding.pry
```
Comment (rizidoro) [inline spec/exact_target_endpoint_spec.rb]: OOps... binding.pry

File: exact_target_endpoint.rb (hunk 1 above)
Comment (jao) [inline exact_target_endpoint.rb]: nope, this comes from the original fork, I thought about removing this, but I keep planning to ask for a honeybadger account, to monitor the integrations.

File: lib/exact_target/triggered_send.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      @subject = email[:subject]
+      @template = email[:template]
+      @variables = email[:variables].to_h.merge('Subject' => subject)
+
+      raise InvalidArguments, "'to', 'template', 'subject', 'variables' attributes are required" if to.blank? || template.blank? || subject.blank? || variables.empty?
+    end
+
+    def send_email
+      trigger = FuelSDK::TriggeredSend.new
+      trigger.client = client
```
Comment (alxpborges) [inline lib/exact_target/triggered_send.rb]: this client comes from superclass?

Comment (jao) [inline lib/exact_target/triggered_send.rb]: yes sir, it's defined in the `ExactTarget::Base.initialize`

https://github.com/petlove/exact_target_integration/blob/deploy_with_data_extension/lib/exact_target/base.rb#L5-L9

**Comment by alxpborges** [comment]
Seems nice!

**Comment by jao** [comment]
let's see if the auto deploy works :dart:

**Pull request merged**